compiler/semantic/scope.go

```diff
@@ -24,31 +30,72 @@ func (s *Scope) Exit() {
 	s.stack = s.stack[:len(s.stack)-1]
 }
 
-func (s *Scope) Bind(name string, ref dag.Op) {
-	s.tos().Define(name, ref)
+func (s *Scope) DefineVar(name string) (*dag.Var, error) {
+	b := s.tos()
+	if _, ok := b.symbols[name]; ok {
+		return nil, fmt.Errorf("symbol %q redefined", name)
+	}
+	ref := &dag.Var{
+		Kind: "Var",
+		Slot: s.nvars(),
+	}
+	b.Define(name, ref)
+	b.nvar++
+	return ref, nil
 }
 
-func (s *Scope) Lookup(name string) dag.Op {
+func (s *Scope) DefineConst(name string, def dag.Expr) error {
+	b := s.tos()
+	if _, ok := b.symbols[name]; ok {
+		return fmt.Errorf("symbol %q redefined", name)
+	}
+	val, err := kernel.EvalAtCompileTime(s.zctx, def)
+	if err != nil {
+		return err
+	}
+	if val.IsError() {
+		return fmt.Errorf("cannot resolve const %q at compile time", name)
```

Comment:
Should we include `zson.MustFormatValue(*val)` in this error?

Reply:
It's an error so better to just take the error string.